python/cudf/cudf/tests/test_string.py

```diff
@@ -52,6 +53,15 @@ def ps_gs(data, index):
     return (ps, gs)
 
 
+@pytest.mark.parametrize("nbytes", [0, 2 ** 10, 2 ** 31 - 1, 2 ** 31, 2 ** 32])
+@patch.object(nvstrings.nvstrings, "byte_count")
+def test_from_nvstrings_nbytes(mock_byte_count, nbytes):
+    mock_byte_count.return_value = nbytes
+    expectation = raise_builder([nbytes > np.iinfo(np.int32).max], MemoryError)
```

Comment:
generally a good idea to keep logic out of unit tests. one test for "size too big" and another for "size too small" would indicate by glance _which_ of the `nbytes` are acceptable and which are not.

Reply:
I do see the merits of breaking this test into two separate tests. On the other hand, that will lead to code repetition.

Testing for errors conditionally in a parametrized test is a pattern described in the pytest docs: http://doc.pytest.org/en/latest/example/parametrize.html#parametrizing-conditional-raising

Reply:
It looks like that document `parametrize`s expectations along with the respective arguments, but that's different than what is being done here, correct?

I don't feel strongly either way in this case, but I do like what the article suggests.

Reply:
Yup. I can definitely put the expectation next to each parametrized value if that sounds better, but honestly I think that is needless repetition. It also puts the burden on the programmer to calculate which parameters will raise and which will not, rather than the computer..